imap_processing/cdf/config/imap_hi_global_cdf_attrs.yaml

```diff
@@ -0,0 +1,24 @@
+# YAML entries at the root level will overwrite the default global attributes
+Data_version: "001"
+Descriptor: &desc
+  Hi>IMAP High-Energy (IMAP-Hi) Energetic Neutral Atom Imager
+TEXT: &txt >
+  IMAP-Hi consists of two identical, single-pixel high-energy energetic
+  neutral atom (ENA) imagers mounted at fixed angles of 90 and 45 degrees
+  relative to the spacecraft spin axis. These imagers measure neutral atoms
+  entering our solar system from the outer edge of the heliosphere as they
+  move towards the Sun. See https://imap.princeton.edu/instruments/imap-hi for
+  more details.
+Instrument_type: Particles (space)
+
+imap_hi_l1a_de_attrs:
+    Data_level: 1A
+    Data_type: L1A_DE>Level-1A Direct Event
+    Logical_source: imap_hi_l1a_{sensor}-de
+    Logical_source_description: IMAP-Hi Instrument Level-1A Direct Event Data.
+
+imap_hi_l1b_de_attrs:
+    Data_level: 1B
```

Comment:
```suggestion
# YAML entries at the root level will overwrite the default global attributes
instrument_base: &instrument_base
  Data_version: "001"
  Descriptor: &desc
    Hi>IMAP High-Energy (IMAP-Hi) Energetic Neutral Atom Imager
  TEXT: &txt >
    IMAP-Hi consists of two identical, single-pixel high-energy energetic
    neutral atom (ENA) imagers mounted at fixed angles of 90 and 45 degrees
    relative to the spacecraft spin axis. These imagers measure neutral atoms
    entering our solar system from the outer edge of the heliosphere as they
    move towards the Sun. See https://imap.princeton.edu/instruments/imap-hi for
    more details.
  Instrument_type: Particles (space)

imap_hi_l1a_de_attrs:
  <<: *instrument_base
  Data_level: 1A
  Data_type: L1A_DE>Level-1A Direct Event
  Logical_source: imap_hi_l1a_{sensor}-de
  Logical_source_description: IMAP-Hi Instrument Level-1A Direct Event Data.

imap_hi_l1b_de_attrs:
  <<: *instrument_base
  Data_level: 1B
```

Reply:
I started out with an implementation much like your suggestion and was having trouble figuring out why my `Data_version` string from this YAML was an integer when I retrieved it from the CDFManager. I found that the data manager code loops over each schema item and first checks if that key is in the global attributes before looking in the instrument specific global attributes. This meant that I was always getting the value back from the `Data_version` defined in the imap_default_global_cdf_attrs.yaml rather than the Hi specific `Data_version` defined in the `instrument_base`. By defining these values at the root level, when the CDFManager loads the file, the values overwrite the values in the global file which means I always get back the values that I want. To view the logic that I am talking about, see this function: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/cdf/cdf_attribute_manager.py#L147

I didn't follow the bug that you and Laura found that caused the move to the `instrument_base` definition style, so I can't speak to that.

Reply:
@maxinelasp will be opening PR to handle this. You and her caught this bug in time.